packages/block-editor/src/components/block-inspector/index.js

```diff
@@ -231,6 +237,23 @@ const BlockInspector = ( { showNoBlockSelectedMessage = true } ) => {
 			/>
 		);
 	}
+
+	if (
+		isOffCanvasNavigationEditorEnabled &&
+		( blockType.name === 'core/navigation' ||
+			blockType.name === 'core/navigation-link' ||
+			blockType.name === 'core/navigation-submenu' )
+	) {
+		return (
+			<NavigatorProvider initialPath={ `/${ selectedBlockClientId }` }>
+				<NavigationInspector
+					selectedBlockClientId={ selectedBlockClientId }
+					blockName={ blockType.name }
+				/>
```

Comment:
I kind of expected a loop at this level to render a "screen" per clientId

Reply:
I took that approach in an earlier commit, but because of the way the provider works, the state wasn't getting updated properly in the screen components. Separating them like this seems to solve the issue.

Reply:
> I took that approach in an earlier commit, but because of the way the provider works, the state wasn't getting updated properly in the screen components.

Why do you mean by "the state wasn't getting updated properly"?

Reply:
Here I have an explanatory video. I'm happy to explain further if you'd like.

https://user-images.githubusercontent.com/5360536/208503788-5ff731de-a146-4c30-acf1-6710f5cd0f59.mov